### PR TITLE
feat(screen): Add scroll tracking support for animated headers

### PR DESCRIPTION
**Pull request by damassi**
Addresses [MOPLAT-810] 

### Description

This adds more robust support for animated headers by updating a few things:
- Renames `AnimatedTabsHeader` to `AnimatedHeader` so that its generalized 
- Adds support for non-tabs animated headers by creating a new `Screen.ScrollView` and `Screen.FlatList`
- Shares main screen tracking logic between the tabs and the above. 
- Added stories

Example of how to use:

#### Screen (`ScrollView` or `FlatList`) 
```tsx
<Screen>
  <Screen.AnimatedHeader title="Hey There" />
  <Screen.Body>
    <Screen.ScrollView>...</Screen.ScrollView>
  </Screen.Body>
</Screen>
```

For our `<Tabs>` component, the only update is to rename `<Screen.AnimatedTabsHeader>` to `<Screen.AnimatedHeader>`

https://github.com/artsy/palette-mobile/assets/236943/11d11035-a3a6-4686-bbcb-9dfa5ccab0dc

cc @artsy/mobile-platform 

[MOPLAT-810]: https://artsyproduct.atlassian.net/browse/MOPLAT-810?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/elements/Screen/index.tsx [hunk 1]
```diff
@@ -4,18 +4,27 @@ import { BottomView } from "./BottomView"
...
 import { ScreenBase } from "./ScreenBase"
+import { ScreenFlatList } from "./ScreenFlatList"
+import { ScreenScrollContextProvider, useScreenScrollContext } from "./ScreenScrollContext"
+import { ScreenScrollView } from "./ScreenScrollView"
 
 export * from "./constants"
 
 export const Screen = Object.assign(ScreenBase, {
-  AnimatedTabsHeader,
+  AnimatedHeader,
```
Comment (damassi) [inline src/elements/Screen/index.tsx:16]: Renamed this, hence the breaking major change

**Review by brainbicycle — APPROVED** [review]
Wooo! Looks great 🙌

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This PR resolves [] <!-- eg [PROJECT-XXXX] -->
+
+### Description
+
+<!-- Info, implementation, how to get there, before & after screenshots & videos, follow-up work, etc -->
```
Comment (damassi) [inline docs/pull_request_template.md:5]: Also added a simple PR template

**Comment by artsyit** [comment]
:rocket: PR was released in [`v11.0.40`](https://github.com/artsy/palette-mobile/releases/tag/v11.0.40) :rocket:

**Comment by damassi** [comment]
ugh! our "Version: Major" labels aren't working. That explains a few things. Will delete them in a sec in favor of the ones that normally ship with `auto`

File: docs/pull_request_template.md (hunk 2 above)
Comment (brainbicycle) [reply docs/pull_request_template.md:5]: nice

**Pull request merged**